### PR TITLE
[WD-13865] redesign UI

### PR DESCRIPTION
**Pull request by akbarkz**
## Done

- Redesigned according to the wireframe in Figma: https://www.figma.com/design/uVXFi46C7ZgriDoWEnsv1U/24.10-Content-system?node-id=16-2320&t=B6Cu2fHvqikmi2b0-4
- Added breadcrumbs
- Created a separate SiteSelector component to select a project
- Fixed bugs related to routing and state

## QA

- Clone the repo
- Run `dotrun --release 1.1.0-rc1 build` to build the app
- Run `dotrun --release 1.1.0-rc1 serve` to run it locally
- Open `0.0.0.0:8104` locally in your browser
- Check the UI and components and see if all works as expected

## Fixes

 - Fixes https://warthogs.atlassian.net/browse/WD-13865

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.

### CODE REVIEW

File: static/client/pages/Webpage/Webpage.tsx [hunk 1]
```diff
@@ -1,33 +1,57 @@
...
       <div>
-        <label htmlFor="page-copy-doc">Copy doc:&nbsp;</label>
-        {page.link ? (
-          <a href={page.link} id="page-link" rel="noopener noreferrer" target="_blank">
-            {page.link}
-          </a>
-        ) : (
-          <input defaultValue={page.link || "-"} id="page-copy-doc" type="text" />
-        )}
+        <p aria-label="page-descr">DESCRIPTION</p>
```
Comment (mtruj013) [inline static/client/pages/Webpage/Webpage.tsx:51]: Same as above
```suggestion
        <p aria-labelledby="page-descr" class="p-text--small-caps">Description</p>
```

File: static/client/components/Breadcrumbs/Breadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        setBreadcrumbs(paths);
+      }
+    } else {
+      setBreadcrumbs([]);
+    }
+  }, [location]);
+
+  return breadcrumbs.map((bc, index) => (
+    <React.Fragment key={`bc-${index}`}>
+      {index < breadcrumbs.length - 1 ? <a href={bc.link}>{bc.name}</a> : <span>{bc.name}</span>}
```
Comment (mtruj013) [inline static/client/components/Breadcrumbs/Breadcrumbs.tsx:33]:
```suggestion
      {index < breadcrumbs.length - 1 ? <a href={bc.link} class="p-text--small-caps">{bc.name}</a> : <span class="p-text--small-caps">{bc.name}</span>}
```

**Review by mtruj013 — CHANGES_REQUESTED** [review]
This looks awesome, exciting stuff!! Found a few bugs and needed changes

File: static/client/components/Navigation/Navigation.tsx [hunk 3]
```diff
@@ -27,16 +32,23 @@ const Navigation = (): JSX.Element => {
...
               <NavigationBanner />
               <div className="l-navigation__controls">
                 <NavigationCollapseToggle isCollapsed={isCollapsed} setIsCollapsed={setIsCollapsed} />
               </div>
+              <SiteSelector />
             </div>
             <div className="p-panel__content">
               <NavigationItems />
             </div>
+            <div className="p-panel__footer">
```
[the remark this answers is not in the document]
Comment (akbarkz) [reply static/client/components/Navigation/Navigation.tsx:46]: nope, this is out of scope, but we'll add it soon too

File: static/client/components/Navigation/Navigation.tsx [hunk 4]
```diff
@@ -27,16 +32,23 @@ const Navigation = (): JSX.Element => {
       </header>
       <nav aria-label="main" className={classNames("l-navigation", { "is-collapsed": isCollapsed })} role="navigation">
```
[the remark this answers is not in the document]
Comment (akbarkz) [reply static/client/components/Navigation/Navigation.tsx:33]: > when the browser is taken off fullscreen (even when not greatly reduced) the navigation panel is halfway hidden
Indeed, that doesn't look nice, but I only got a design for a big screen and unfortunately I don't know how the collapsed sidebar should look like in our case. This is something I am going to ask @juanruitina about.

> the file list is hidden from the panel unless you hover over it
That's weird, I couldn't reproduce this on my machine. What browser and version are you using?

> when selecting a different site from the drop down, if you hover over the un-selected site, the option field for the site you're currently changes to a white background which hides the text
Same here, I couldn't reproduce it with 3 different browsers on my machine: Chrome, Firefox and Safari.

> this one's more a question than a bug, but why make users toggle the base folder themselves once they've selected a site? Wouldn't it be better to have it open by default on selection?
This is actually a bug and it supposed to keep any selected project open by default. Fixed it 👍

[the remark this answers is not in the document]
Comment (mtruj013) [reply static/client/components/Navigation/Navigation.tsx:33]: > Indeed, that doesn't look nice, but I only got a design for a big screen and unfortunately I don't know how the collapsed sidebar should look like in our case

Even for big screens though this is still not collapsing correctly. As soon as I take my screen off of fullscreen, even if the size difference is the smallest unit available so still considered desktop sized screen, the sidebar is toggled which should not be the functionality. This is an [existing Vanilla component ](https://vanillaframework.io/docs/patterns/navigation#side-navigation) (unless I'm misunderstanding something) so I imagine it should work in the same way

**Comment by juanruitina** [comment]
I couldn't make the demo work locally, can you provide a screenshot/full-screen demo video? I've come up with some more designs on how I think the responsiveness should work; I'll be checking with the app layout colleagues if that's consistent with how they use the side navigation.

**Pull request merged**